### ISSUE TITLE
Hit score is NaN

### ISSUE THREAD

**Issue by Zombobot1** [LyraSearch/lyra#215]
**Describe the bug**

Unexpected search results: NaNs and 0

**To Reproduce**
```ts
const db = create({
  schema: {
    id: 'string',
    text: 'string',
  },
})

// silly sentences
const sentences = ['I love my new pets.', 'I give my love to your pets.', 'Every child loves my ice cream.']
sentences.forEach((sentence, i) => insert(db, { id: i.toString(), text: sentence.toLowerCase() }))

const r = search(db, {
  term: 'i m',
  properties: ['text'],
})
console.log(r) // 3 hits, all with NaN score
```

Technically, it's correct but it has zero score

```ts
// initialization is same as above

const r = search(db, {
  term: 'i',
  properties: ['text'],
})
console.log(r) // 3rd match (ice cream) has 0 as score
```

**Expected behavior**
Scores are meaningful

**Additional context**
All versions are the latest, tested in chrome.

**Comment by micheleriva** [comment]
Hi there,
I'm not saying that this is acceptable/reasonable, but the reason why you're getting `0` (but please let me know if you also get `NaN`, I couldn't reproduce this) is that you don't have enough data in your db. If you try with a bigger dataset, it shouldn't be a problem.

**Comment by Zombobot1** [comment]
I don't mind about 0 but I don't like getting NaNs. This is a [sandbox](https://codesandbox.io/s/objective-lamport-m0fxqc?file=/src/index.js) with reproduction

**Comment by jinh0** [comment]
I'm also having the issue of the score being NaN.

**Comment by jinh0** [comment]
From inspecting the source code, the NaN issue might be coming from line 706 of `lyra.ts`:

```javascript
const prevScore = uniqueDocsIDs.get(id)!;
uniqueDocsIDs.set(id, prevScore + tfIdfScore);
```
The ! ignores the fact that .get can give us `undefined`, and turns out when you add `undefined` to a number, as in `prevScore + tfldfScore`, you can get `NaN`.

**Comment by toomuchdesign** [comment]
The issue was addressed in [this PR](https://github.com/LyraSearch/lyra/pull/193) and should available in next Lyra release.

**Comment by micheleriva** [comment]
Thanks @toomuchdesign, I forgot about this one. Closing this issue, will be fixed in `v0.4.0`
